src/mon.c

```diff
@@ -540,7 +540,8 @@ make_corpse(struct monst *mtmp, unsigned int corpseflags)
         }
         goto default_1;
     case PM_LONG_WORM:
-        (void) mksobj_at(WORM_TOOTH, x, y, TRUE, FALSE);
+        otmp = mksobj_at(WORM_TOOTH, x, y, TRUE, FALSE);
+        free_oname(otmp); /* remove potential weapon_oname */
```

Comment:
Rather than fixing up the oname after the fact, worm teeth should either be excluded in the if statement in mksobj determining if weapon_oname gets called at all, or excluded at the top of weapon_oname itself. I can't think of a reason why any worm teeth should get these names.